NewForce-Cohort-9/tabloid-full-stack-code-crusaders
# Added Reaction Features to View and Add

# Description 
Added View Reactions function for a post: [Trello Ticket #35](https://trello.com/c/reWYNT3J)
Added Add Reaction function: [Trello Ticket #34](https://trello.com/c/ztXZBpcw)

# Type of Change 
Added new, non-breaking feature

## Testing Instructions for Viewing Reactions

1. git add and git commit whatever you're working on on your branch
2. git checkout main
3. git fetch --all
4. git checkout cb-branch5
5. If you don't see the most recent version of my files, git pull origin cb-branch5

6. Insert table for Reaction into VS Community: 
`SET IDENTITY_INSERT Reaction ON
INSERT INTO Reaction (Id, [Name], ImageLocation) VALUES
(1, 'Like', 'https://cdn-icons-png.flaticon.com/512/25/25297.png'),
(2, 'Dislike', 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ6Jt4fWTTKq-a3g4LAk1FNBRURO87dt5UDjg&s');
SET IDENTITY_INSERT Reaction OFF` 

7. Then Insert table for PostReaction:
`INSERT INTO PostReaction (PostId, ReactionId, UserProfileId)
VALUES
(1, 1, 1),
(1, 2, 2),
(1, 1, 3);`

8. Check to make sure the tables where input correctly use this SQL:
`SELECT * FROM PostReaction;`

9. In VS Community, click the execute with debugger solid green button to launch the application
10. In your GitBash Terminal, cd into tabloid-full-stack-code-crusaders/TabloidFullStack/TabloidFullStack/client/tabloid and run: npm start
11. This should redirect you back to the Login page.
12. Log in as user `foo@bar.com`
13. This should take you to the home screen where you will see the navbar and a text greeting:
14. Welcome to Tabloid, Foo!
15. Click on the option `Posts` in the navbar
16. Click the post called `morph front-end markets` to view details
17.See `Reaction` buttons with count of the thumbs up (2) and thumbs down (1)

## Testing Instructions to Add a Reaction

1. Click `Return to Posts`
2. Scroll to find post called `drive front-end portals` click post to view details
3. Click the like `thumbs up` reaction and see the count go up to (1)
4. Go into your VS Community and re-run the SQL `SELECT * FROM PostReaction;`
5. You should see a new addition to the table with a PostId: 8 ReactionId: 1 UserProfileId:1

# Checklist:
